### PR TITLE
broadcast: reject truncated cycle points

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Closes https://github.com/cylc/cylc-flow/issues/6407

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] Changelog entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/cycling/__init__.py [hunk 1]
```diff
@@ -91,8 +91,16 @@ def _cmp(self, other) -> int:
         """Compare self to other point, returning a 'cmp'-like result."""
         pass
 
-    def standardise(self) -> 'PointBase':
-        """Format self.value into a standard representation and check it."""
+    def standardise(self, allow_truncated: bool = True) -> 'PointBase':
+        """Format self
+        value into a standard representation and check it.
```
Comment (oliver-sanders) [inline cylc/flow/cycling/__init__.py:96]:
```suggestion
        """Format self.value into a standard representation and check it.
```

**Review by wxtim — COMMENTED** [review]
Looks sane. Have tried it and it appears to fix the bug in question.

Presumably waiting for release stuff to finish before adding a changelog?

**Comment by oliver-sanders** [comment]
> Presumably waiting for release stuff to finish before adding a changelog?

Waiting for a PR number!

**Review by MetRonnie — APPROVED** [review]
This also prevents traceback when doing e.g. `cylc hold <future-task>` using a truncated cycle

**Pull request merged**